File: backend/src/main/java/com/greenteam/FullStackApplication/dtos/FullUserDto.java
package com.greenteam.FullStackApplication.dtos;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@NoArgsConstructor
@Data
public class FullUserDto {
    private Long id;

    private ProfileDto profile;

    private boolean admin;

    private boolean active;

    private String status;

    private Set<CompanyDto> companies;

    private Set<TeamDto> teams;

    public Set<CompanyDto> getCompanies(){
        return this.companies;
    }
}

File: frontend/src/app/models/FullUserDTO.ts
import ProfileDTO from "./ProfileDTO";
import CompanyDTO from "./CompanyDTO";
import TeamDTO from "./TeamDTO";
export default interface FullUserDTO {
    id: number,
    profile: ProfileDTO,
    isAdmin: boolean,
    active: boolean,
    status: string,
    companies: CompanyDTO[],
    teams: TeamDTO[]
};